### ISSUE TITLE
'substring(with:)' is deprecated: Please use String slicing subscript. (Swift 4)

### ISSUE THREAD

**Issue by aornano** [scinfu/SwiftSoup#32]
There are some errors in the new Xcode 9 (Swift 4) for **String.swift** file, that can be solved easily (I've commented the old code..) :

```
func indexOf(_ substring: String, _ offset: Int ) -> Int {
        if(offset > characters.count) {return -1}

        let maxIndex = self.characters.count - substring.characters.count
        if(maxIndex >= 0) {
            for index in offset...maxIndex {
                let rangeSubstring = self.characters.index(self.startIndex, offsetBy: index)..<self.characters.index(self.startIndex, offsetBy: index + substring.characters.count)
                if self[rangeSubstring] == substring {
                //if self.substring(with: rangeSubstring) == substring {
                    return index
                }
            }
        }
        return -1
    }
```
and:

```
static func split(_ value: String, _ offset: Int, _ count: Int) -> String {
        let start = value.index(value.startIndex, offsetBy: offset)
        let end = value.index(value.startIndex, offsetBy: count+offset)
        let range = start..<end
        //return value.substring(with: range)
        return String(value[range])
    }
```

**Comment by 0xTim** [comment]
Fixed in #35

**Comment by scinfu** [comment]
Thank you Tim
